Stripe/STPPaymentCardTextField.m

```diff
@@ -718,8 +719,23 @@ - (CGFloat)numberFieldFullWidth {
 }
 
 - (CGFloat)numberFieldCompressedWidth {
-    // Current longest possible longest pan fragment is 5 characters
-    return [self widthForText:@"88888"];
+
+    NSString *cardNumber = self.cardNumber;
+    if (cardNumber.length == 0) {
+        cardNumber = self.viewModel.defaultPlaceholder;
+    }
+
+    STPCardBrand currentBrand = [STPCardValidator brandForNumber:cardNumber];
+    NSArray<NSNumber *> *sortedCardNumberFormat = [[STPCardValidator cardNumberFormatForBrand:currentBrand] sortedArrayUsingSelector:@selector(unsignedIntegerValue)];
+    NSUInteger fragmentLength = [STPCardValidator fragmentLengthForCardBrand:currentBrand];
+    NSUInteger maxLength = MAX([[sortedCardNumberFormat lastObject] unsignedIntegerValue], fragmentLength);
+
+    NSMutableString *maxCompressedString = [[NSMutableString alloc] initWithCapacity:maxLength];
+    for (NSUInteger i = 0; i < maxLength; ++i) {
+        [maxCompressedString appendString:@"8"];
```

Comment:
Looks like you can do this in a one-liner with

```objective-c
NSString *maxCompressedString = [@"" stringByPaddingToLength:maxLength withString:@"8" startingAtIndex:0];
```

I had no idea that method existed, and this is *totally* fine with me as-is. I was curious if foundation provided something we could use and now that I've found it, wanted to share.

Reply:
oh same. Cool!